challenges-services/src/main/java/org/exoplatform/challenges/service/ChallengeServiceImpl.java

```diff
@@ -18,15 +25,18 @@
 
   private ChallengeStorage challengeStorage;
 
-  private SpaceService     spaceService;
+  private AnnouncementStorage announcementStorage;
+
+  private SpaceService spaceService;
 
-  public ChallengeServiceImpl(ChallengeStorage challengeStorage, SpaceService spaceService) {
+  public ChallengeServiceImpl(ChallengeStorage challengeStorage, SpaceService spaceService, AnnouncementStorage announcementStorage) {
     this.challengeStorage = challengeStorage;
     this.spaceService = spaceService;
+    this.announcementStorage = announcementStorage;
   }
 
   @Override
-  public Challenge createChallenge(Challenge challenge, String username) throws IllegalAccessException {
+  public ChallengeRestEntity createChallenge(Challenge challenge, String username) throws IllegalAccessException {
```

Comment:
The same as for Announcements, we need to let the Rest entity be in the Rest services layer.